agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/HttpServerTracer.java

```diff
@@ -195,6 +205,48 @@ protected void onRequest(final Span span, final REQUEST request) {
     // TODO set resource name from URL.
   }
 
+  protected void onConnectionAndRequest(Span span, CONNECTION connection, REQUEST request) {
+    String flavor = flavor(connection, request);
+    if (flavor != null) {
+      SemanticAttributes.HTTP_FLAVOR.set(span, flavor);
+    }
+    SemanticAttributes.HTTP_CLIENT_IP.set(span, clientIP(connection, request));
+  }
+
+  protected String clientIP(CONNECTION connection, REQUEST request) {
+    // try Forwarded
+    String forwarded = requestHeader(request, "Forwarded");
+    if (forwarded != null) {
+      // may be split by ; or ,
+      for (String forwardedValues : FORWARDED_SPLIT_PATTERN.split(forwarded)) {
+        forwardedValues = forwardedValues.trim();
+        String[] split = EQUALS_PATTERN.split(forwardedValues, 2);
+        if (split.length != 2) {
+          continue;
+        }
+        if ("for".equalsIgnoreCase(split[0]) && !split[1].isEmpty()) {
+          return split[1];
+        }
+      }
+    }
+
+    // try X-Forwarded-For
+    forwarded = requestHeader(request, "X-Forwarded-For");
+    if (forwarded != null) {
+      // may be split by ,
+      int endIndex = forwarded.indexOf(',');
+      if (endIndex > 0) {
+        forwarded = forwarded.substring(0, endIndex);
```

Comment:
did we need trim here?
```suggestion
        forwarded = forwarded.substring(0, endIndex).trim();
```

Reply:
no, there aren't going to be any spaces before the first comma and we only need the first value